requests/maniphest_search.go

```diff
@@ -0,0 +1,116 @@
+package requests
+
+import (
+	"encoding/json"
+	"errors"
+	"github.com/uber/gonduit/entities"
+)
+
+// ManiphestSearchRequest represents a request to maniphest.search API method.
+type ManiphestSearchRequest struct {
+	// QueryKey is builtin or saved query to use. It is optional and sets initial constraints.
+	QueryKey    string             `json:"queryKey,omitempty"`
+	// Constraints contains additional filters for results. Applied on top of query if provided.
+	Constraints *SearchConstraints `json:"constraints,omitempty"`
+	// Attachments specified what additional data should be returned with each result.
+	Attachments *SearchAttachments `json:"attachments,omitempty"`
+
+	*entities.Cursor
+	Request
+}
+
+// SearchAttachments contains fields that specify what additional data should be returned with sesarch results.
+type SearchAttachments struct {
+	// Subscribers if true instructs server to return subscribers list for each task.
+	Subscribers bool `json:"subscribers,omitempty"`
+	// Columns requests to get the workboard columns where an object appears.
+	Columns     bool `json:"columns,omitempty"`
+	// Projects requests to get information about projects.
+	Projects    bool `json:"projects,omitempty"`
+}
+
+// ManiphestRequestSearchOrder describers how results should be ordered.
+type ManiphestRequestSearchOrder struct {
+	// Builtin is the name of predefined order to use.
+	Builtin string
+	// Order is list of columns to use for sorting, e.g. ["color", "-name", "id"],
+	Order   []string
+}
+
+// UnmarshalJSON parses JSON  into an instand of ManiphestRequestSearchOrder type.
+func (o *ManiphestRequestSearchOrder) UnmarshalJSON(data []byte) error {
+	if o == nil {
+		return errors.New("maniphest search order is nil")
+	}
+	if jerr := json.Unmarshal(data, &o.Builtin); jerr == nil {
+		return nil
+	}
+
+	return json.Unmarshal(data, &o.Order)
+}
+
+// MarshalJSON creates JSON our of ManiphestRequestSearchOrder instance.
+func (o *ManiphestRequestSearchOrder) MarshalJSON() ([]byte, error) {
+	if o == nil {
+		return nil, errors.New("maniphest search order is nil")
+	}
+	if o.Builtin != "" {
+		return json.Marshal(o.Builtin)
+	}
+	if len(o.Order) > 0 {
+		return json.Marshal(o.Order)
+	}
+
+	return nil, nil
+}
+
+// SearchConstraints describes search criteria for request.
+type SearchConstraints struct {
+	// IDs - search for objects with specific IDs.
+	IDs            []string        `json:"ids,omitempty"`
```

Comment:
Must be `[]int`.

Reply:
The reason I use string here is because of https://github.com/uber/gonduit/blob/master/requests/maniphest_gettasktransactions.go#L5 - it is also string there.
I don't use IDs in my search queries so this has not been tested. I did an ad-hoc test to use []int here and it seems to be fine.

Reply:
I noticed this too, but it looks like Phabricator is a bit liberal with
accepting `[]string` despite the API saying it's `[]int`. Since the rest of
the library uses `[]string` for IDs, I don't feel strongly about this one way
or another.

Reply:
`*.search` methods do not accept strings for INT values - you'll get an error.

Actually I'd recommend perform real requests with different parameters to Phabricator and double check if no errors are thrown. Please ping me after you'll test so I'll be able to merge this PR. Thanks!